### PR TITLE
Fix nodemailer, add nodemailer-ses-transport, improve aws-sdk.

### PR DESCRIPTION
**Pull request by westy92**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

Description of changes:
- fixed nodemailer so it would compile locally.
- combined nodemailer to be one file so it works with `typings`.
- added tempate methods to nodemailer along with tests.
- added nodemailer-ses-transport and tests.
- improved aws-sdk.

### CODE REVIEW

File: aws-sdk/aws-sdk.d.ts [hunk 1]
```diff
@@ -355,10 +355,10 @@ declare module "aws-sdk" {
       ConditionalOperator?: "AND" | "OR";
       Expected?: {
         [someKey: string]: {
-          AttributeValueList: any[];
-          ComparisonOperator: _DDBDC_ComparisonOperator;
+          AttributeValueList?: any[];
+          ComparisonOperator?: _DDBDC_ComparisonOperator;
           Exists: boolean;
```
Comment (westy92) [inline aws-sdk/aws-sdk.d.ts]: If Exists is false, the other attributes will not be specified.

File: nodemailer/nodemailer-types.d.ts [hunk 2]
```diff
@@ -1,149 +0,0 @@
-﻿// Type definitions for Nodemailer 1.3.2
```
Comment (westy92) [inline nodemailer/nodemailer-types.d.ts]: I combined this file with nodemailer.d.ts.  Typings needed this to all be in one file.

**Comment by westy92** [comment]
Bump

File: aws-sdk/aws-sdk.d.ts (hunk 1 above)
Comment (sandersn) [inline aws-sdk/aws-sdk.d.ts]: It would be good to note that in a jsdoc comment.

**Pull request merged**